netlink-packet-route/src/rtnl/route/message.rs

```diff
@@ -40,3 +43,98 @@ impl<'a, T: AsRef<[u8]> + 'a> Parseable<RouteMessageBuffer<&'a T>> for Vec<Nla>
         Ok(nlas)
     }
 }
+
+fn octets_to_addr(octets: &[u8]) -> Result<IpAddr, DecodeError> {
+    if octets.len() == 4 {
+        let mut ary: [u8; 4] = Default::default();
+        ary.copy_from_slice(octets);
+        Ok(IpAddr::from(ary))
+    } else if octets.len() == 16 {
+        let mut ary: [u8; 16] = Default::default();
+        ary.copy_from_slice(octets);
+        Ok(IpAddr::from(ary))
+    } else {
+        Err(DecodeError::from("Cannot decode IP address"))
+    }
+}
+
+impl RouteMessage {
+    /// Returns the input interface index, if present.
+    pub fn input_interface(&self) -> Option<u32> {
+        self.nlas.iter().find_map(|nla| {
+            if let Nla::Iif(v) = nla {
+                Some(*v)
+            } else {
+                None
+            }
+        })
+    }
+
+    /// Returns the output interface index, if present.
+    pub fn output_interface(&self) -> Option<u32> {
+        self.nlas.iter().find_map(|nla| {
+            if let Nla::Oif(v) = nla {
+                Some(*v)
+            } else {
+                None
+            }
+        })
+    }
+
+    /// Returns the source address prefix, if present.
+    pub fn src_prefix(&self) -> Option<(IpAddr, u8)> {
```

Comment:
I have a slight preference for `source` instead of `src`, and `destination` or `dest` instead of `dst`.